### PR TITLE
gsoc: searchdocs: Add searchdocs plugin

### PR DESCRIPTION
**Pull request by meetmangukiya**
Closes https://github.com/coala/corobo/issues/15

# Reviewers Checklist

- [ ] Appropriate logging is done.
- [ ] Appropriate error responses.
- [ ] Handle every possible exception.
- [ ] Make sure there is a docstring in the command functions. Hint: Lookout for
  `botcmd` and `re_botcmd` decorators.
- [ ] See that 100% coverage is there.
- [ ] See to it that mocking is not done where it is not necessary.

### CODE REVIEW

File: plugins/searchdocs.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+class Searchdocs(BotPlugin):
+    """
+    Search API and user docs
+    """
+
+    API_DOCS = 'http://api.coala.io/en/latest'
+    USER_DOCS = 'http://docs.coala.io/en/latest'
+
+    @botcmd
+    def search(self, msg, arg):
```
Comment (Udayan12167) [inline plugins/searchdocs.py]: Add doc comment on how it searches.

Comment (Udayan12167) [inline plugins/searchdocs.py]: The doc comment should certainly contain the correct inputs expected by this method. As in intended behavior in order to be able to evaluate the regexes.

Comment (meetmangukiya) [inline plugins/searchdocs.py]: All of them are provided implicitly by the errbot

**Comment by Udayan12167** [comment]
ack https://github.com/coala/corobo/pull/100/commits/fe10f1cdd2788a3549b1cbd78c19d6ff0d46e48c

**Comment by meetmangukiya** [comment]
reack fe10f1cdd2788a3549b1cbd78c19d6ff0d46e48c

**Comment by meetmangukiya** [comment]
ack fe10f1cdd2788a3549b1cbd78c19d6ff0d46e48c

**Comment by meetmangukiya** [comment]
@rultor merge

**Comment by meetmangukiya** [comment]
reack 0934a6964806d2767f6b272b77d7dbbe97c8ff4b

**Comment by meetmangukiya** [comment]
ack 0934a6964806d2767f6b272b77d7dbbe97c8ff4b

**Comment by meetmangukiya** [comment]
ack 0934a69

**Comment by meetmangukiya** [comment]
@rultor merge

**Pull request merged**